src/test/java/test/InvokedMethodNameListener.java

```diff
@@ -68,8 +68,6 @@ public void afterInvocation(IInvokedMethod method, ITestResult testResult) {
         if (!(skipConfiguration && method.isConfigurationMethod())) {
           if (wantSkippedMethodAfterInvocation) {
             skippedAfterInvocationMethodNames.add(name);
-          } else {
-            throw new IllegalStateException("A skipped test is not supposed to be invoked");
```

Comment:
Yes this is expected because we are now ensuring that for all use-cases the `afterInvocation()` is invoked for skipped tests as well. Currently it was getting invoked when a test gets skipped due to a config failure, but it wasn't getting invoked when a parent test fails. So had to remove this edit check in this listener to facilitate that. The other option I had was to basically create another listener, which I wasn't very comfortable with.

Reply:
I'm not feeling it good because we already changed the behavior twice.
It looks users want both behaviors.
Maybe it should be configurable. WDYT?

Reply:
@juherr - I dont mind putting in a configuration flag (Let me guess.. You want it via `org.testng.internal.Configuration#Configuration()` right :) ).

But here's the deal. TestNG is already exhibiting inconsistency with respect to `afterInvocation()`.

1. When there's a skip caused by a configuration failure, we call `beforeInvocation()/afterInvocation()`
2. When there's a skip caused by a failure via `dependsOnMethods/dependsOnGroups` we don't call `beforeInvocation()/afterInvocation()`

With this changeset, I have fixed this anomaly and ensured that `beforeInvocation()/afterInvocation()` gets called for all skips in `@Test` methods. That is why I had to remove this exception.

With respect to having a flag that controls if `beforeInvocation()/afterInvocation()` should be called or not for test methods, I would prefer to have it done via a separate PR.